ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction_service.proto

```diff
@@ -128,6 +135,46 @@ message GetFlatTransactionResponse {
   Transaction transaction = 1;
 }
 
+message GetContractByIdRequest {
+  // The contract id being queried
+  // Required
+  string contract_id = 1;
+}
+message GetContractByIdResponse {
+  // The collection of events, will only contain consumed ExerciseEvents
+  repeated TreeEvent events = 1;
+}
+
+message GetContractByKeyRequest {
+
+  // The contract key to search for.  Note that only contracts that are visible to the
+  // requesting parties will returned.  To ensure that no contract key assignments are missed
```

Comment:
I am not sure this needs to be guaranteed on the grpc interface. The requirements here are more relaxed than for the command interface. I can well imagine that users want to request all key-related operations visible to them and are not concerned with those they cannot see due to missing authorization.

Reply:
Oscurity is security ;-)